sdks/python/apache_beam/utils/pipeline_options.py

```diff
@@ -20,6 +20,69 @@
 import argparse
 
 from apache_beam.transforms.display import HasDisplayData
+from apache_beam.utils.value_provider import StaticValueProvider
+from apache_beam.utils.value_provider import RuntimeValueProvider
+from apache_beam.utils.value_provider import ValueProvider
+
+
+def static_value_provider_of(value_type):
+  """"Helper function to plug a ValueProvider into argparse.
+
+  Args:
+    value_type: the type of the value. Since the type param of argparse's
+                add_argument will always be ValueProvider, we need to
+                preserve the type of the actual value.
+  Returns:
+    A partially constructed StaticValueProvider in the form of a function.
+
+  """
+  def _f(value):
+    _f.func_name = value_type.__name__
+    return StaticValueProvider(value_type, value)
+  return _f
+
+
+class ValueProviderArgumentParser(argparse.ArgumentParser):
+  """This class provides an API to add options of ValueProvider type.
+
+  It preserves the functionalities of the parent ArgumentParser.
```

Comment:
Since all users needs to subclass it, whether they want to use templates or not this comments needs to reflect that.

Similarly class name should be generic. This is the only way to define arguments not only value provider arguments.